Added CMake build system

alexey-lysiuk opened the pull request:
Build game and tools, the latter can be disabled
Search for SDL2, paths can be confiigured
Detect MMX support in target compiler

alexey-lysiuk commented:
I can update Travis CI configuration to use CMake.
This allows to build with several versions of GCC and Clang on Linux, and to add macOS target in the future.

Also, if you are interested I can add AppVeyor too. You need to enable it first for your repository.

Panzerschrek commented on PanzerChasm/FindSDL2.cmake:
```diff
@@ -0,0 +1,173 @@
+
+	# For MinGW library
+	IF(MINGW)
+		SET(SDL2_LIBRARY_TEMP ${MINGW32_LIBRARY} ${SDL2_LIBRARY_TEMP})
```
This brokes my build with MinGW.

> C:/lnk/SDL2-2.0.3/lib/x86/SDL2main.lib(./Release/SDL_windows_main.obj):(.text[_main]+0x0): multiple definition of `main'
> C:/Qt5.3.0/Tools/mingw482_32/bin/../lib/gcc/i686-w64-mingw32/4.8.2/../../../../i686-w64-mingw32/lib/../lib/libmingw32.a(lib32_libmingw32_a-crt0_c.o):crt0_c.c:(.text.startup+0x0): first defined here

alexey-lysiuk replied:
Works fine for me using MinGW 4.8.2 installed with the latest Qt Creator. SDL version is 2.0.7 though.
What are the values of SDL2_LIBRARY and SDL2MAIN_LIBRARY variables in your CMake build configuration?

Panzerschrek replied:
Oh, crap.
I used SDL2 for Visual Studio. With SDL2 for MinGW all works fine.

Panzerschrek commented on PanzerChasm/CMakeLists.txt:
```diff
@@ -0,0 +1,262 @@
+
+CHECK_CXX_SOURCE_COMPILES("#include <mmintrin.h>
+    int main(void) { __m64 v = _m_from_int(0); }"
+    HAVE_MMX)
```
Replace spaces with tabs here and below.

Panzerschrek reviewed (commented):
All, except linking problem on MinGW is fine.
Tomorrow i will check build also on GNU/Linux.

alexey-lysiuk commented:
Fixed spaces and switched to CMake in Travis configuration.
I’ll made the single clean commit later.

alexey-lysiuk committed:
Added CMake build system

Build game and tools, the latter can be disabled
Search for SDL2, paths can be confiigured
Detect MMX support in target compiler

alexey-lysiuk commented:
I've pushed clean version.
